string resourceName
uint32 requestedQuantity
